### DISCUSSION TITLE
images not working in Github

### DISCUSSION THREAD

**Discussion by MerelTimmer** [community/community#54525] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hello,

I'm new to Github and have to use it for a school assignment. I have uploaded my work to Github but I'm facing an issue here. All the images that I have called in CSS using "url(/images/go-to-jail.png);" are not visible on Github. 

I have already checked for capital letters and typos but I can't find anything. 

When inspecting, the console also gives the following error message:

ps://mereltimmer.github.io/VIDii/styles/images/go-to-jail.png 404

Here is my repo:
https://github.com/MerelTimmer/VIDii

And the page:
https://mereltimmer.github.io/VIDii/

Does someone know how can I solve this problem?

**Comment by airtower-luna** [comment]
You have two kinds of URLs for the images in your CSS, both invalid in different ways. To quote two examples:

```css
cursor: url(/images/cursorhat.png) 55 15, auto;
background-image: url(images/go-to-jail.png);
```

The first one is an *absolute* URL, so the browser looks for it right under the hostname (so `https://mereltimmer.github.io/images/cursorhat.png`). That doesn't work because your site is deployed in a subdirectory. You could make it work by including the `/VIDii/` part, but there's a better option.

The second one is a *relative* URL, which means the browser will search for the file starting from the directory where the file containing the URL (your stylesheet) is. This is generally the right approach, but the path needs to be correct. Your `images/` directory isn't within the `styles/` one, so you need to go one level up (to the root directory of your repository) first:

```css
background-image: url('../images/go-to-jail.png');
```

See also: https://developer.mozilla.org/en-US/docs/Learn/Common_questions/Web_mechanics/What_is_a_URL#absolute_urls_vs._relative_urls

**Reply by MerelTimmer** [reply]
This works indeed. I do understand it a little, but for now I'm glad this is the solution.

There are only some images still not working, but I saw on other discussions that Github needs some time to upload al the images, right?

**Reply by airtower-luna** [reply]
> There are only some images still not working, but I saw on other discussions that Github needs some time to upload al the images, right?

Deployment can take some time, but I'd expect the images to be deployed together with the updated CSS, so I don't think that's the issue here. You'll have to look at the individual images that still have issues to see what might be wrong (could something else with the path, or other issues). If you need help figuring it out, best post a link right to the line with the problem.
